src/test/java/com/github/sdonkov/PrefixTreeTest.java

```diff
@@ -3,28 +3,42 @@
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 
-
 import java.io.IOException;
 import java.net.URISyntaxException;
 import java.nio.file.Files;
 import java.nio.file.Paths;
+import java.util.HashSet;
 import java.util.List;
 
-import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 
 public class PrefixTreeTest {
 
     private static List<String> words;
-    private static PrefixTree prefixTree;
+    private static PrefixTree<String> prefixTree;
 
     @BeforeAll
     static void setUp() throws IOException, URISyntaxException {
-        prefixTree = new PrefixTree();
+        prefixTree = new PrefixTree<>();
         words = Files.readAllLines(Paths.get(PrefixTreeTest.class.getResource("/dictionary.txt").toURI()));
         words.forEach(word -> prefixTree.add(word));
```

Comment:
It'd be nice to start using the newly introduced `addAll` here, I guess. Same for the other lambdas - `containsAll` should be able to replace them probably.